File: app/src/main/java/ru/iandreyshev/parserrss/ui/adapter/FeedTabsAdapter.java
package ru.iandreyshev.parserrss.ui.adapter;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

import ru.iandreyshev.parserrss.app.Utils;
import ru.iandreyshev.parserrss.models.rss.ViewRss;
import ru.iandreyshev.parserrss.ui.fragment.FeedPageFragment;

public class FeedTabsAdapter extends FragmentStatePagerAdapter {
    private final List<ViewRss> mRssList = new ArrayList<>();

    public FeedTabsAdapter(final FragmentManager manager) {
        super(manager);
    }

    public void insert(final ViewRss rss) {
        mRssList.add(rss);
        notifyDataSetChanged();
    }

    public void remove(final ViewRss rss) {
        mRssList.remove(rss);
        notifyDataSetChanged();
    }

    public ViewRss getRss(int position) {
        if (position < 0 || position >= mRssList.size()) {
            return null;
        }

        return mRssList.get(position);
    }

    @Override
    public Fragment getItem(int position) {
        return FeedPageFragment.newInstance(getRss(position));
    }

    @Override
    public int getCount() {
        return mRssList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return Utils.truncateTabsTitle(getRss(position).getTitle());
    }

    @Override
    public int getItemPosition(@NonNull final Object item) {
        int position = mRssList.indexOf(item);

        if (position < 0) {
            return POSITION_NONE;
        }

        return position;
    }
}

File: app/src/main/java/ru/iandreyshev/parserrss/ui/adapter/FeedTabsAdapter.kt
package ru.iandreyshev.parserrss.ui.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import ru.iandreyshev.parserrss.R
import ru.iandreyshev.parserrss.app.App

import java.util.ArrayList

import ru.iandreyshev.parserrss.models.rss.ViewRss
import ru.iandreyshev.parserrss.ui.extention.tabTitle
import ru.iandreyshev.parserrss.ui.fragment.FeedPageFragment

class FeedTabsAdapter(manager: FragmentManager) : FragmentStatePagerAdapter(manager) {
    private val rssList = ArrayList<ViewRss>()
    val isEmpty
        get() = rssList.isEmpty()

    fun insert(rss: ViewRss) {
        rssList.add(rss)
        notifyDataSetChanged()
    }

    fun remove(rss: ViewRss) {
        rssList.remove(rss)
        notifyDataSetChanged()
    }

    fun getRss(position: Int): ViewRss? {
        return rssList.getOrNull(position)
    }

    override fun getItem(position: Int): Fragment {
        val rss = getRss(position)

        when (rss) {
            null -> throw IllegalArgumentException("Invalid item position")
            else -> return FeedPageFragment.newInstance(rss)
        }
    }

    override fun getCount(): Int {
        return rssList.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return getRss(position)?.title?.tabTitle ?: App.getStr(R.string.feed_tab_title)
    }

    override fun getItemPosition(item: Any): Int {
        val position = rssList.indexOf(item)

        return if (position < 0) POSITION_NONE else position
    }
}
